src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -88,6 +101,26 @@ public boolean publish(String message, String color) {
         return result;
     }
 
+    private String getTokenToUse() {
+        if (credentialsAuthTokenId != null && !credentialsAuthTokenId.isEmpty()) {
+            StringCredentials credentials = lookupCredentials(credentialsAuthTokenId);
+            if (credentials != null) {
+                logger.info("Using secret for Slack auth token.");
+                return credentials.getSecret().getPlainText();
+            }
+        }
+
+        logger.info("Using plain text for Slack auth token.");
+
+        return token;
+    }
+
+    private StringCredentials lookupCredentials(String credentialId) {
```

Comment:
I'm not sure how feasible it is to use, but it looks as though this whole method could be replaced by a single call to `CredentialsProvider.findCredentialById`